Added paragraph about the usage of rnaseq with prokaryotic data based on Issue #765 (#790)
nf-core/rnaseq: docs/usage.md

* The `--gtf_group_features_type` parameter will automatically be set to `gene_type` as opposed to `gene_biotype`, respectively.
* If you are running Salmon, the `--gencode` flag will also be passed to the index building step to overcome parsing issues resulting from the transcript IDs in GENCODE fasta files being separated by vertical pipes (`|`) instead of spaces (see [this issue](https://github.com/COMBINE-lab/salmon/issues/15)).

## Adapt pipeline parameters for prokaryotes

> The default settings of the pipeline are mainly adapted for eukaryotes but have to be changed slightly for prokaryotes. The main reason for this is the different genetic architecure of prokaryotes. The below mentioned parameters work if a `gff` file is provided as reference. 

Reviewer: ```suggestion
The default settings of the pipeline are for eukaryotes but have to be changed slightly for prokaryotes. The main reason for this is the different genetic architecture of prokaryotes. The below mentioned parameters work if a `gff` file is provided as reference. 
```

Changes and parameter specifications for prokaryotes:

> * `--featurecounts_feature_type transcript` since the default value `exon` does not contain the required `--featurecounts_group_type gene_biotype` specification.

Author: If you check the gtf file, __exon__ rows don't contain a 'gene_biotype' in the last column. Only the __transcript__ rows have this information. My suggestion is as you mention above *--featurecounts_feature_type "transcript" --featurecounts_group_type "gene_biotype"*. Since the second part (...group_type) is already the default setting you don't need to include it in the command.

> * `--featurecounts_feature_type transcript` since the default value `exon` does not contain the required `--featurecounts_group_type gene_biotype` specification.

Reviewer: I see, you are right. Could it be that this varies? I thought I did use recently (with the old rnaseq version) "transcript" but there was still no "gene_biotype".

My point here is that "exon" is never a good choice for bacterial RNA-Seq. In the files you linked, "exon" is present 121 times but "transcript" 4417 times in the gtf file. In the source gff file its 121 "exon" but 4298 CDS (which might be also an option) and 4359 "gene".

> * You can use `--featurecounts_feature_type CDS` in combination with `--featurecoutns_group_type product` but than featureCounts will no longer reflect the biotypes of your RNA. It could be helpful to identify the number of hypothetical proteins. 

Reviewer: ```suggestion
* `--featurecounts_feature_type "CDS" --featurecoutns_group_type "product"` but than featureCounts will no longer reflect the biotypes of your RNA. It could be helpful to identify the number of hypothetical proteins. 
```

> * If your execution struggle with Salmon as aligner, change `--alginer` to hisat2.

Reviewer: ```suggestion
* `--aligner "hisat2"` in case the default (`"star_salmon"`) fails.
```
Do you have any explanation why that happens? In regards to prokaryotic genomes?

Author: Not exactly. With my data, which I used during the week of the hackathon, all aligner worked but for my previous tries I remember that only hisat2 worked. Maybe it is due to some assumptions for the quantification, I haven't looked into the details of the aligners.

Reviewer: Alright, thanks.

> * You can skip RSeQC with `--skip_rseqc` since it mainly focus on eukaryotic features like splice junctions, transcription start (TSS) and ending sites (TES)

Reviewer: ```suggestion
* `--skip_rseqc` skip RSeQC since features like splice junctions, transcription start (TSS) and ending sites (TES) are less informative in prokaryotes than in eukaryotic.
```

> * If you aren't interested in the biotypes of your RNA data, you can skip the whole process with `--skip_biotype_qc`

Reviewer: ```suggestion
* `--skip_biotype_qc` in case biotypes of your RNA data are of no interest.
```

> > **NB:** For older versions of the pipeline the names may be different. Check the paramters docs for details. 

Reviewer: I am actually not sure thats needed, I'd omit it.
